src/syntax/syntax.rkt

```diff
@@ -116,6 +116,13 @@
     (table-set! operators 'operator
                 (make-hash (list (cons 'type type) (cons 'args args) (cons 'key value) ...)))))
 
+(define (no-complex fun)
+  (λ (x . xs)
```

Comment:
Why specialize the first argument this way? If you want to allow any arguments, you write `(lambda xs <body>)`.